sencha-workspace/SlateDemonstrationsStudent/app/view/Dashboard.js

```diff
@@ -149,11 +156,24 @@ Ext.define('SlateDemonstrationsStudent.view.Dashboard', {
         me.setRecentProgress(contentAreaSet);
         me.setLegend(contentAreaSet);
         me.setCardsCt(contentAreaSet);
+        me.setNonEnrollmentMessage(contentAreaSet);
         Ext.resumeLayouts(true);
 
         me.fireEvent('selectedcontentareachange', me, contentArea, oldContentArea);
     },
 
+    updateCardsForEnrollmentStatus: function(enrolled) {
```

Comment:
@BillClinton so, final nitpick. `update*` and `apply*` function names on components are used by the config system and should be avoided for methods that aren't config lifecycles.

That said, I was thinking this earlier, this is actually a good case for using the config system to mediate state changes. Put `hasEnrollments: true` up above and then rename this to `updateHasEnrollments` to match, and then the call from the outside becomes `component.setHasEnrollments(trueOrFalse)` -- the benefit there then is that the config system handles only calling `update*` when the value changes. So calling `component.setHasEnrollments(true)` after the last call was `component.setHasEnrollments(true)` won't run the `update` method a second time. It's a nice little mechanic that lets you "assert" the desired state aggressively and then let the config system handle debouncing and only running the update code when there's a change

If making a config didn't make sense, I'd typically prefix a method like this with `sync*`

Reply:
👍